packages/authorization/src/authorize-interceptor.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/authorization
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  asGlobalInterceptor,
+  bind,
+  filterByTag,
+  inject,
+  Interceptor,
+  Provider,
+} from '@loopback/context';
+import * as debugFactory from 'debug';
+import {getAuthorizeMetadata} from './decorators/authorize';
+import {
+  AuthorizationContext,
+  AuthorizationDecision,
+  AuthorizeFn,
+} from './types';
+
+const debug = debugFactory('loopback:authorization:interceptor');
+
+@bind(asGlobalInterceptor('authorization'))
+export class AuthorizationInterceptor implements Provider<Interceptor> {
+  constructor(
+    @inject(filterByTag('authorizationProvider'))
+    private authorizationFunctions: AuthorizeFn[],
+  ) {}
+
+  value(): Interceptor {
+    return async (invocationCtx, next) => {
+      const description = debug.enabled ? invocationCtx.description : '';
+      const metadata = getAuthorizeMetadata(
+        invocationCtx.target,
+        invocationCtx.methodName,
+      );
+      if (!metadata) {
+        debug('No authorization metadata is found %s', description);
+        return await next();
+      }
+      debug('Authorization metadata for %s', description, metadata);
+      const user = await invocationCtx.get<{name: string}>('current.user', {
```

Comment:
@bajtos This is the same concern as https://github.com/strongloop/loopback-next/issues/2246, I use `UserProfile` defined in `@loopback/authentication` instead of `{name: strign}` to make it at least consistent with the authentication module, and let's address the flexibility issue in story https://github.com/strongloop/loopback-next/issues/2900